libwaysip/src/dispatch.rs

```diff
@@ -0,0 +1,326 @@
+
+use crate::state::{self, LayerSurfaceInfo, SecondState};
+use wayland_client::protocol::wl_output;
+use wayland_client::{
+    delegate_noop,
+    globals::{GlobalListContents},
+    protocol::{
+        wl_buffer::WlBuffer,
+        wl_compositor::WlCompositor,
+        wl_keyboard, wl_pointer, wl_registry,
+        wl_seat::{self},
+        wl_shm::{WlShm},
+        wl_shm_pool::WlShmPool,
+        wl_surface::WlSurface,
+    },
+    Connection, Dispatch, Proxy, WEnum,
+};
+
+use wayland_protocols::xdg::shell::client::{xdg_toplevel::XdgToplevel, xdg_wm_base};
+
+use wayland_protocols::xdg::xdg_output::zv1::client::{
+    zxdg_output_manager_v1::ZxdgOutputManagerV1, zxdg_output_v1,
+};
+
+use wayland_protocols_wlr::layer_shell::v1::client::{
+    zwlr_layer_shell_v1::{ZwlrLayerShellV1},
+    zwlr_layer_surface_v1::{self},
+};
+
+use wayland_protocols::wp::cursor_shape::v1::client::{
+    wp_cursor_shape_device_v1::{self, WpCursorShapeDeviceV1},
+    wp_cursor_shape_manager_v1::WpCursorShapeManagerV1,
+};
+
```

Comment:
extra empty line

Reply:
There's a lot more of those, will be removed once it's no longer a draft